### PR TITLE
use statefulset instead of deployment to implement singleton worker and server

### LINKED ISSUE #279: Singleton of OAM is replicas=1 in K8s StatefulSet
In [OAM spec](https://github.com/microsoft/hydra-spec/blob/master/3.component_model.md#singleton-server) we defined:

_A Singleton Server is a special case of the Server workload type that is limited to **at most one** replica._

Normally, we could achieve Singleton in K8s by using a ReplicaSet (Deployment) or StatefulSet with replicas=1. 

A ReplicaSet favors availability and prioritizes keeping a single instance up (**at least one** semantics). That could occasionally lead to more than one instance running at the same time, for example when a node is disconnected from the cluster and the ReplicaSet starts another pod on a different node without confirming that the first pod is stopped. 

A StatefulSet favors consistency over availability, and provides **at most one** semantics. In the case of a node being disconnected from the cluster, it won’t start the pod on a healthy node. For example, K8s CSI plugin [use this pattern](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md#cluster-level-deployment).

In that case, we may want to use StatefulSet instead of Deployment in Scylla.

/cc @wonderflow @technosophos

### PR DESCRIPTION
**Pull request by wonderflow**
fixes #279

### CODE REVIEW

File: src/schematic/scopes/health.rs [hunk 1]
```diff
@@ -74,12 +74,22 @@ impl Health {
         params: Vec<ParameterValue>,
     ) -> Result<Self, Error> {
         let probe_method = match extract_string_params("probe-method", params.clone()) {
-            Some(network_id) => network_id,
-            None => return Err(format_err!("probe-method is not exist")),
+            Some(probe_method) => probe_method,
+            None => {
+                return Err(format_err!(
+                    "probe-method is not exist in params {:?}",
```
Comment (technosophos) [inline src/schematic/scopes/health.rs]:
```suggestion
                    "probe-method does not exist in params {:?}",
```

File: src/schematic/scopes/health.rs [hunk 2]
```diff
@@ -74,12 +74,22 @@ impl Health {
...
+                ))
+            }
         };
         let probe_endpoint = match extract_string_params("probe-endpoint", params.clone()) {
-            Some(network_id) => network_id,
-            None => return Err(format_err!("probe-endpoint is not exist")),
+            Some(probe_endpoint) => probe_endpoint,
+            None => {
+                return Err(format_err!(
+                    "probe-endpoint is not exist in params {:?}",
```
Comment (technosophos) [inline src/schematic/scopes/health.rs]:
```suggestion
                    "probe-endpoint does not exist in params {:?}",
```

**Comment by wonderflow** [comment]
@technosophos  thanks!

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**